config_ext_test.go

```diff
@@ -0,0 +1,426 @@
+		assert.Equal(t, tt.Interface, tt.Value.Value(), "unexpected result from Value")
+		require.NoError(t, tt.Value.Populate(tt.Populate), "error populating")
+	}
+
+	t.Run("missing", func(t *testing.T) {
+		var s string
+
+		t.Run("top level", func(t *testing.T) {
+			run(TestCase{
+				Value:     provider.Get("not_a_key"),
+				HasValue:  false,
+				Interface: nil,
+				Populate:  &s,
+			})
+			assert.Equal(t, "", s, "unexpected result after Populate")
+		})
+
+		t.Run("subkey", func(t *testing.T) {
+			run(TestCase{
+				Value:     provider.Get("practical.cadillac"),
+				HasValue:  false,
+				Interface: nil,
+				Populate:  &s,
+			})
+			assert.Equal(t, "", s, "unexpected result after Populate")
+		})
+
+		t.Run("descend into slice", func(t *testing.T) {
+			run(TestCase{
+				Value:     provider.Get("fun.extra"),
+				HasValue:  false,
+				Interface: nil,
+				Populate:  &s,
+			})
+			assert.Equal(t, "", s, "unexpected result after Populate")
+		})
+	})
+
+	t.Run("nil", func(t *testing.T) {
+		var s string
+		run(TestCase{
+			Value:     provider.Get("nothing"),
+			HasValue:  true,
+			Interface: nil,
+			Populate:  &s,
+		})
+		assert.Equal(t, "", s, "unexpected result after Populate")
+	})
+
+	t.Run("scalar", func(t *testing.T) {
+		var s string
+		run(TestCase{
+			Value:     provider.Get("practical.toyota"),
+			HasValue:  true,
+			Interface: "camry",
+			Populate:  &s,
+		})
+		assert.Equal(t, "camry", s, "unexpected result after Populate")
+	})
+
+	t.Run("map", func(t *testing.T) {
+		var m map[string]string
+		run(TestCase{
+			Value:    provider.Get("practical"),
+			HasValue: true,
+			Interface: map[interface{}]interface{}{
+				"toyota": "camry",
+				"honda":  "civic",
+				"nissan": "altima",
+			},
+			Populate: &m,
+		})
+		assert.Equal(t, map[string]string{
+			"toyota": "camry",
+			"honda":  "civic",
+			"nissan": "altima",
+		}, m, "unexpected result after populate")
+	})
+
+	t.Run("slice", func(t *testing.T) {
+		var s []string
+		run(TestCase{
+			Value:     provider.Get("fun"),
+			HasValue:  true,
+			Interface: []interface{}{"maserati", "lamborghini"},
+			Populate:  &s,
+		})
+		assert.Equal(t, []string{"maserati", "lamborghini"}, s, "unexpected result after populate")
+	})
+
+	t.Run("struct", func(t *testing.T) {
+		type Occupants struct {
+			Driver    string
+			Passenger string
+			Backseat  []string
+		}
+		var o Occupants
+		run(TestCase{
+			Value:    provider.Get("occupants.honda"),
+			HasValue: true,
+			Interface: map[interface{}]interface{}{
+				"driver":    "jane",
+				"passenger": "arthur",
+				"backseat":  []interface{}{"nora"},
+			},
+			Populate: &o,
+		})
+		assert.Equal(t, Occupants{
+			Driver:    "jane",
+			Passenger: "arthur",
+			Backseat:  []string{"nora"},
+		}, o, "unexpected result after populate")
+	})
+
+	t.Run("nil merged into scalar", func(t *testing.T) {
+		t.Skip("TODO: ignores explicit nil")
```

Comment:
thanks for adding this -- one other case that might be worth adding is nil merging into an existing sequence/map. I'm not sure of the right behaviour, I'd assume replace in both cases, but since maps merge, perhaps it just gets ignored?

Reply:
Good point. I'd expect sequences to be replaced and maps to be merged, since that's the behavior for non-nil things.